pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/embeddings/EmbeddingsService.java

```diff
@@ -0,0 +1,8 @@
+package com.datastax.oss.pulsar.functions.transforms.embeddings;
+
+import java.util.List;
+
+public interface EmbeddingsService {
+
+  List<List<Double>> calculateEmbeddings(List<String> texts);
```

Comment:
"throws" ?

Also ideally this is a async API, currently Transforms aren't supporting async processing so we can keep it this way, but in the future we have to make it fully async, otherwise it is hard to deal with parallelism

Reply:
I think there s no need here for a checked exception for now. In the future we might want to let the user configures the behavior in case of too long inputs. In that case we could implement a specific exception

I'll keep it as sync for now